packages/react/src/components/Inbox.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import type { InboxNotification } from '@novu/js';
+import type { BaseNovuUIOptions } from '@novu/js/ui';
+import { useRenderer } from '../context/RenderContext';
+import { Mounter } from './Mounter';
+import { Renderer } from './Renderer';
+
+type InboxDefaultProps = {
+  renderNotification?: (notification: InboxNotification) => React.ReactNode;
+  renderBell?: ({ unreadCount }: { unreadCount: number }) => React.ReactNode;
+};
+
+const InboxDefault = (props: InboxDefaultProps) => {
+  const { renderNotification, renderBell } = props;
+  const { novuUI, mountElement } = useRenderer();
+
+  const mount = React.useCallback(
+    (element: HTMLElement) => {
+      return novuUI.mountComponent({
+        name: 'Inbox',
+        props: {
+          renderNotification: renderNotification
+            ? (el, { notification }) => mountElement(el, renderNotification(notification))
+            : undefined,
+          renderBell: renderBell ? (el, { unreadCount }) => mountElement(el, renderBell({ unreadCount })) : undefined,
+        },
+        element,
+      });
+    },
+    [renderNotification, renderBell]
+  );
+
+  return <Mounter mount={mount} />;
+};
+
+type BaseProps = BaseNovuUIOptions;
+
+type DefaultProps = BaseProps &
+  InboxDefaultProps & {
+    children?: never;
+  };
+
+type WithChildrenProps = BaseProps & {
+  children: React.ReactNode;
+};
+
+type InboxProps = DefaultProps | WithChildrenProps;
+
+export const Inbox = (props: InboxProps) => {
+  if ('children' in props) {
+    const { children, ...options } = props as WithChildrenProps;
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** Can use a [custom type-guard](https://blog.logrocket.com/how-to-use-type-guards-typescript/#:~:text=a%20Boolean%20value.-,Custom%20type%20guards,-TypeScript%20custom%20type) here to avoid type assertion!‏